echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java

```diff
@@ -34,7 +36,18 @@
   private PubsubSubscribers pubsubSubscribers;
 
   @RequestMapping(value = "/pubsub/subscriptions", method = RequestMethod.GET)
-  List<String> getSubscriptions() {
-    return pubsubSubscribers.getAll().stream().map(s -> s.getName()).collect(Collectors.toList());
+  List<PubsubSubscriptionBySystem> getSubscriptions() {
+    return pubsubSubscribers
+        .getAll()
+        .stream()
+        .map(s -> new PubsubSubscriptionBySystem(s.pubsubSystem().toString(), s.getName()))
+        .collect(Collectors.toList());
+  }
+
+  @AllArgsConstructor
+  @Getter
```

Comment:
I think you can omit the `@Getter` here.

Reply:
`@Data` is the right call. Also, I would recommend using a `@Builder` and to construct this class instead of the constructor above. You will need `@AllArgsConstructor` and `@NoArgsConstructor` to get serialization/deserialization to work.